### DISCUSSION TITLE
`cascade.REMOVE` 와 `orphanremoval = true` 의 차이

### DISCUSSION THREAD

**Discussion by gyungchan-jo** [2024-woowacourse-study/level-interview#226] [Spring Data]
.

**Comment by Kimprodp** [comment]
cascade = CascadeType.REMOVE와 orphanRemoval = true는 JPA에서 엔티티 관계를 관리할 때 사용하는 옵션으로
두 설정 모두 부모 엔티티가 삭제되면 자식 엔티티도 함께 삭제되지만, 부모/자식 관계를 제거할 땐 다른 방식으로 동작한다.

- CascadeType.REMOVE : 부모 엔티티가 자식 엔티티와의 관계를 제거해도 자식 엔티티는 삭제되지 않고 그대로 DB에 남아있다.

- orphanRemoval = true : 부모 엔티티가 자식 엔티티의 관계를 제거하면 자식은 고아 객체(부모 엔티티와 연관관계가 끊어진 자식 엔티티)로 취급되어 DB에서 사라진다.

**Comment by seunghye218** [comment]
cascade.REMOVE와 orphanRemoval는 JPA에서 엔티티 간의 관계를 관리할 때 사용되는 옵션입니다. 이 두 가지는 엔티티의 생명 주기를 관리하는 데 중요한 역할을 하지만, 그 목적과 작동 방식에 차이가 있습니다.

- cascade.REMOVE
    - 부모 엔티티가 삭제될 때, 연관된 자식 엔티티도 함께 삭제됩니다.
    - 연관된 모든 자식 엔티티에 대해 삭제 작업이 전파됩니다.
- orphanRemoval
    - 부모 엔티티와의 관계가 끊어진 자식 엔티티를 자동으로 삭제합니다.
    - 부모 엔티티에서 자식 엔티티를 컬렉션에서 제거하면, 해당 자식 엔티티가 데이터베이스에서 삭제됩니다.

cascade.REMOVE는 부모 엔티티가 삭제될 때 자식 엔티티도 함께 삭제하고 싶을 때 유용하며, orphanRemoval은 부모 엔티티에서 자식 엔티티를 제거할 때 자식 엔티티를 데이터베이스에서도 삭제하고자 할 때 사용됩니다.
